packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4559,6 +4564,72 @@ class _UpdateTextSelectionToAdjacentLineAction<T extends DirectionalCaretMovemen
   bool get isActionEnabled => state._value.selection.isValid;
 }
 
+class _UpdateTextSelectionToAdjacentPageAction<T extends DirectionalCaretMovementIntent> extends ContextAction<T> {
+  _UpdateTextSelectionToAdjacentPageAction(this.state);
+
+  final EditableTextState state;
+
+  VerticalCaretMovementRun? _verticalMovementRun;
+  TextSelection? _runSelection;
+
+  void stopCurrentVerticalRunIfSelectionChanges() {
```

Comment:
Can this method be removed?

Reply:
I don't think so: vertical runs should be kept with both arrows and page up/down. I now see though that the call to it is missing in `_didChangeTextEditingValue()`.

I'll experiment with mergint the two Line/Page intents and actions into one to simplify things a bit.